src/vec.rs

```diff
@@ -323,7 +323,7 @@ where
     {
         let mut vec = Vec::new();
         for i in iter {
-            vec.push(i).expect("Vec::from_iter overflow");
+            vec.push(i).ok().expect("Vec::from_iter overflow");
```

Comment:
In the afterthought, this could also be implemented in terms of `Vec::extend` which would reduce code duplication and needed space for the statically allocated except message.